src/qt/bitcoin.cpp

```diff
@@ -211,11 +212,15 @@ BitcoinApplication::~BitcoinApplication()
     delete window;
     window = nullptr;
 #ifdef ENABLE_WALLET
-    delete paymentServer;
-    paymentServer = nullptr;
-    delete m_wallet_controller;
-    m_wallet_controller = nullptr;
-#endif
+    if (paymentServer) {
```

Comment:
This is not necessary, it is safe to `delete nullptr`.

Reply:
Or maybe you want to get rid of these `delete` as these objects are owned by `BitcoinApplication` instance.

Reply:
> Or maybe you want to get rid of these `delete` as these objects are owned by `BitcoinApplication` instance.

Exactly!